bcda/client/bluebutton.go

```diff
@@ -0,0 +1,110 @@
+package client
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+type BlueButtonClient struct {
+	httpClient http.Client
+}
+
+func NewBlueButtonClient() *BlueButtonClient {
+	certFile := os.Getenv("BB_CLIENT_CERT_FILE")
+	keyFile := os.Getenv("BB_CLIENT_KEY_FILE")
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	tlsConfig := &tls.Config{
+		Certificates:       []tls.Certificate{cert},
+		InsecureSkipVerify: true,
```

Comment:
A thing I forgot -- we'll eventually need to provide a [CA cert](https://gist.github.com/michaljemala/d6f4e01c4834bf47a9c4#file-tls-client-go-L28-L39) here as well. Maybe we can check for `BB_CA_CERT_FILE` and modify the `tls.Config` if the file exists/can be loaded? If nothing else, we should create a ticket to make sure this requirement is tracked!

Reply:
Maybe I'm not following, but the BB2 backend doesn't have a CA cert; it's self-signed.
You could use a CA for your client cert(s) if you wanted, but that's not currently a requirement from our side, either.

Now I'm wondering if we should start using an internal CA for these service-to-service client certs. Thoughts? (My current thinking: No, self-signed is probably still okay, maybe even preferable, as I don't think centrally managing them really gets us anything other than hassle & complexity right now.)

Reply:
I probably have my terminology wrong. What I want to accomplish: make sure that the self-signed cert presented by the BB server is the one that BCDA expects/trusts.

I agree with you on client certs -- I don't think there's a compelling reason to manage centrally atm.

Reply:
Added ticket: https://jira.cms.gov/browse/BCDA-336 Would probably benefit from more detail. :)